### ISSUE TITLE
Batched MCMC: padded catalog with no line columns builds a leading-dim mismatch into lax.map

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#1607]
## Summary

**Static analysis, not yet reproduced at runtime** — filing with the exact reading so the probe is cheap.

In `CatalogFitter`'s vmapped MCMC branch, when a catalog carries **no** line columns *and* padding is active, `all_line_flux` / `all_line_err` keep leading dimension `n_gal` while every sibling array is padded to `n_padded`. All eight arrays are then handed to `jax.lax.map`, which maps over axis 0 and requires a common leading dimension.

Found while unifying the per-galaxy channel table in #1605; deliberately *not* refactored over.

## The reading

`src/tengri/inference/catalog_fitter.py`, no-lines case:

```python
else:
    all_line_flux_orig = jnp.zeros((n_gal, 0), dtype=all_data_orig.dtype)
    all_line_err_orig  = jnp.zeros((n_gal, 0), dtype=all_data_orig.dtype)
```

then the padding branch:

```python
if n_pad_extra > 0:
    all_data  = jnp.concatenate([all_data_orig,  jnp.zeros((n_pad_extra, n_data))], axis=0)
    all_noise = jnp.concatenate([all_noise_orig, jnp.ones((n_pad_extra, n_data))],  axis=0)
    ...
    n_line_cols = all_line_flux_orig.shape[1] if all_line_flux_orig.ndim > 1 else 0
    if n_line_cols > 0:
        ...
    else:
        all_line_flux = all_line_flux_orig     # <- stays (n_gal, 0)
        all_line_err  = all_line_err_orig
```

and both land in `xs`:

```python
xs = (all_init, gal_keys, all_data, all_noise,
      all_presence, all_redshift, all_line_flux, all_line_err)
...
all_positions, all_divergent = jax.lax.map(_run_one, xs, batch_size=K)
```

So `all_data` is `(n_padded, n_data)` and `all_line_flux` is `(n_gal, 0)`.

## Reachability

`n_pad_extra = n_padded - n_gal`, and `_resolve_n_padded` pads only to the next multiple of `K`. With the default `forward_chunk_size=1`, `n_padded == n_gal` and `n_pad_extra == 0`, so the branch is never taken — which is presumably why it has not been hit.

It should require all three:

1. `method="mcmc_nuts"` or `"mcmc_hmc"` (the vmapped branch),
2. `forward_chunk_size > 1` (or an explicit `n_pad`) with `n_galaxies` not a multiple of it,
3. a catalog with **no** `line_cols`.

That is an ordinary batched configuration, so if the reading is right this is reachable from normal use.

## Probe

```python
cat = Catalog(fwd, table, flux_unit="cgs_fnu", flux_cols=[...], err_cols=[...])  # no line_cols
cat.fit(method="mcmc_nuts", key=..., forward_chunk_size=4, n_warmup=8, n_samples=8)
# with n_galaxies not a multiple of 4
```

Expected if the reading holds: a shape/leading-dimension error out of `lax.map`.

## Fix sketch

Pad the zero-width line arrays to `n_padded` like every other channel — `jnp.zeros((n_padded, 0))` is well-formed and `lax.map` is happy with a zero-width trailing axis. The `n_line_cols > 0` special case exists only to avoid concatenating an empty block, which is not actually a problem.

The durable version is the follow-up noted in #1605: drive the stack-and-pad step from `GALAXY_CHANNELS` with a per-channel default and pad fill, so all six channels take one code path instead of six hand-written pairs. That removes the class rather than this instance.

Refs #1605, #1599.
